3884 – Improve seeds: create feed

vasconsaurus opened the pull request:
## Description

We are now creating a feed with published items (randomized verified and false items).
Important: It can take a bit of time for the items to appear in the published list and on the feed (and even when they appear, the count on the published list stays at 0)

## Images
<img width="1114" alt="Screenshot 2023-11-13 at 13 25 11" src="https://github.com/meedan/check-api/assets/87862340/b6595f92-e3dc-41e6-898d-a4e938204b9f">
<img width="1466" alt="Screenshot 2023-11-10 at 13 37 55" src="https://github.com/meedan/check-api/assets/87862340/f5fdaa0c-2152-4072-a0cd-eea520216bc7">

References: 3384

## How has this been tested?

Ran seed script, created and added to user.

## Things to pay attention to during code review

1. When verifying and publishing fact checks we get a lot of logs, I couldn't figure out how to pause logging for the seed file.
2. I decided to create everything with channel: Whatsapp, instead of duplicating everything

## Documentation
I will update the documentation to include:
1. running with elasticsearch_log set to 0, so it doesn't output the logs
    a. `elasticsearch_log=0 bundle exec rake db:seed`
3. Things might take sometime to update in the UI, but if there are any issues with the Feed not displaying the clickable items (items with `published_article_url`), one should log into the rails console, clear the cache, and refresh the browser
    b. `Rails.cache.clear`

vasconsaurus commented:
@brianfleming can you take a look at this one? It's just about creating the populated feed.

vasconsaurus commented:
>However, when I look at some of the items in the list, a lot of them don't have fact-checks, but everything in the shared feed shows content (where is that coming from?) is this related to the delay you mention in the main comment?

Not all of the items have fact checks, but the ones that are in the feed are the ones that do (since they have to be verified and published). (I'm not sure if I understood your question)

>none of the fact-checks in the list have links. It would be a bit more realistic if some of them has a destination article

That might have to do with the delay, could you double check? They seem to be working fine for me. (famous last words)

<img width="1162" alt="Screenshot 2023-11-13 at 17 23 05" src="https://github.com/meedan/check-api/assets/87862340/74bd71d0-20ad-4c06-84a5-7352ec2b6e7d">
<img width="1161" alt="Screenshot 2023-11-13 at 17 26 18" src="https://github.com/meedan/check-api/assets/87862340/6a600361-cbc2-4b5d-8f33-1735cf67cbc6">

brianfleming commented:
@vasconsaurus I am re-running the whole script fresh, just to make sure of things, so its taking a while as you mentioned.

Meanwhilte, the link. I meant this link:
<img width="359" alt="Screenshot 2023-11-13 at 4 20 54 PM" src="https://github.com/meedan/check-api/assets/418001/08aae69e-d248-4be6-9f5c-3f0648720b54">

The shared feed list page behaves differently if the fact-check has an article link or not, this was the minor thing I mentioned i wasnt sure if some of the fact-checks included links

vasconsaurus commented:
@brianfleming 

> The shared feed list page behaves differently if the fact-check has an article link or not, this was the minor thing I mentioned i wasnt sure if some of the fact-checks included links

Ah, I see! Indeed we aren't adding any article links. Will add some 🙂

brianfleming commented:
@vasconsaurus I wiped everything and ran the seed script from nothing. Here is what I am seeing that causes me confusion:

#### The created list has 15 published items in it: good
<img width="720" alt="Screenshot 2023-11-14 at 9 22 27 AM" src="https://github.com/meedan/check-api/assets/418001/78702856-e79e-4528-9676-afcab059a804">

#### The shared feed shows 15 items
<img width="1959" alt="Screenshot 2023-11-14 at 9 19 56 AM" src="https://github.com/meedan/check-api/assets/418001/96e60c9b-227e-4905-96b3-86534ebaa92c">

As you can see a lot of the fact-check content is not showing up, curious as to why.

As I look through the items that are going into the shared feed, I see things in states that I think are incorrect

#### This one appears as unpublished, but somehow is published, and no fact-check content?
<img width="991" alt="Screenshot 2023-11-14 at 9 20 54 AM" src="https://github.com/meedan/check-api/assets/418001/103ecde3-7928-442c-8cf3-eeacb212ca75">

#### I don't think a suggested item should be published? or available to a shared feed?
<img width="756" alt="Screenshot 2023-11-14 at 9 21 05 AM" src="https://github.com/meedan/check-api/assets/418001/15400bc2-c6fe-40c3-8b34-bdc2e2b5c6a0">

vasconsaurus commented:
@brianfleming 
The items not showing happened to me, but was one of those things that after some time were working.
I'll look into and fix the rest

melsawy commented on db/seeds.rb:
```diff
@@ -65,9 +74,29 @@ def create_description(project_media)
+  project_medias[0..9].each do |pm|
+    status = ['verified', 'false'].sample
+
+    verification_status = pm.get_dynamic_annotation('verification_status')
+    verification_status.set_fields = { verification_status_status: status }.to_json
+    verification_status.reload
+    verification_status.save!
```
For simplicity we can use this code to update status
```
verification_status = pm.last_status_obj
verification_status.status = status
verification_status.save!
```

vasconsaurus replied:
thanks @melsawy ❤️

vasconsaurus commented:
@brianfleming 
I fixed (🤞 ) the state issues, and added the published article url. Could you take a look again? See if there are still any issues?

And again, it can take a while for everything to work as expected: 
1. for the published articles to show as published when you search for them
     a. for my last run with everything it took around 26 minutes
2. for them to show in the feed
3. for the feed to differentiate between items that have the url and the ones that don't 
    a. for all my runs but the last one, it took a long time after the feed was done, for my last run it still isn't working, so if it does not update for you, don't worry, I'm still double checking this one

I'll talk to Caio to see if any of this can be faster.

brianfleming commented:
> @brianfleming I fixed (🤞 ) the state issues, and added the published article url. Could you take a look again? See if there are still any issues?
> 
> And again, it can take a while for everything to work as expected:
> 
> 1. for the published articles to show as published when you search for them
>    a. for my last run with everything it took around 26 minutes
> 2. for them to show in the feed
> 3. for the feed to differentiate between items that have the url and the ones that don't
>    a. for all my runs but the last one, it took a long time after the feed was done, for my last run it still isn't working, so if it does not update for you, don't worry, I'm still double checking this one
> 
> I'll talk to Caio to see if any of this can be faster.

I can keep waiting on this, but I rand the script a while ago (when i merged in develop)...and I am still not seeing links.

### For example, White-Jerde, in the shared feed is not clickable (meaning no link)
<img width="680" alt="Screenshot 2023-11-16 at 3 05 42 PM" src="https://github.com/meedan/check-api/assets/418001/9e74ca83-e882-4e1c-9b69-bf335bc1e7ce">

### But the fact-check does have a link on the item page:
<img width="410" alt="Screenshot 2023-11-16 at 3 05 47 PM" src="https://github.com/meedan/check-api/assets/418001/78e574bd-ab56-44e2-a0c0-f23b0e9f39a5">

vasconsaurus commented:
@brianfleming 
I talked to Caio, and turns out it's a cache issue. 🙃 
What worked for me was running `Rails.cache.clear` inside the rails console, and then refreshing the page

brianfleming commented:
> Rails.cache.clear

NIIIICE straight 💸 ...that was supposed to be cash, stupid emoji

anyway, thanks @vasconsaurus that worked for me as well.

cc: thanks to @caiosba for the back-from-vacation-tips-and-tricks

brianfleming commented:
ps. I also appreciate that the fact-check link i clicked on, took me here:

https://www.thespruceeats.com/step-by-step-basic-cake-recipe-304553

caiosba reviewed (approved):
Very good, Manu! Thanks for applying the things we discussed yesterday.
